### PR TITLE
feat: add FontAwesome, header navbar RWD, footer, home page revamp

### PR DESCRIPTION
**Pull request by flynnhou**
## Types of Changes

**Thanks for sending a pull request! Please fill in the following content to let us know better about this change.**
Please put an `x` in the box that applies.

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

### FontAwesome

Install the packages to display the icons in the footers and various usages.

### Header navbar RWD

1. Align the style with the design
2. Toggle different style of the header among devices resolutions
3. Refactor `ExtLink` and `LocaleLink` to align with the design
4. Create `NavBarHamburger` and `NavBarItemAccordion` to align with the design
  - note that the right-hand-side drawer behaves slightly differently for faster development

### Footer

Use the FontAwesome icons along with the RWD

### Home page revamp

- Show the hero image and conference information
- Refactor `TextButton` component to replace the redundant styling in the home page 

## Steps to Test This Pull Request

2. Click on '...'
3. Scroll down to '...'

## Related Issue

If applicable, reference to the issue related to this pull request.

## More Information

**Screenshots**
image
image
image

image
image
image

**Additional context**
Add any other context about the problem here. You may also want to refer
to [how to write the perfect pull request](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/).

### CODE REVIEW

File: components/core/footer/Footer.vue [hunk 1]
```diff
@@ -1,52 +1,70 @@
...
                     href="mailto:organizers@pycon.tw"
-                    target="_blank"
-                    class="outline-none"
-                >
-                    Email
-                </a>
+                    :icon="['fas', 'envelope-square']"
+                />
+                <footer-icon
+                    href="https://tw.pycon.org/2020/zh-hant/ext/discord/"
```
Comment (mattwang44) [inline components/core/footer/Footer.vue:45]: Let's replace the URL of the old instruction page with the invitation link of the `Python Taiwan` discord server. 

```suggestion
                    href="https://discord.com/invite/94hgCQv"
```

File: components/core/footer/Footer.vue [hunk 2]
```diff
@@ -1,52 +1,70 @@
...
+            <div class="w-full flex justify-center items-center flex-wrap">
+                <footer-icon
+                    href="https://pycontw.blogspot.tw/"
+                    :icon="['fab', 'blogger']"
+                />
+                <footer-icon
+                    href="https://www/facebook.com/pycontw"
+                    :icon="['fab', 'facebook']"
+                />
+                <footer-icon href="" :icon="['fab', 'telegram']" />
```
Comment (mattwang44) [inline components/core/footer/Footer.vue:31]: Add the invitaion link of telegram channel.
```suggestion
                <footer-icon 
                    href="https://t.me/pycontw_chat" 
                    :icon="['fab', 'telegram']" 
                />
```

File: pages/index.vue [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+            </div>
+            <div class="page-homeLandingFocus__image mx-12 md:mx-0">
+                <img src="/key-vision-logo.png" alt="" />
+            </div>
+            <div
+                class="page-homeLandingFocus__context flex flex-col text-golden text-center md:text-left mb-4 ml-0 md:mb-0 md:ml-8"
+            >
+                <div class="font-serif">
+                    <div class="text-3xl lg:text-5xl">PyCon Taiwan 2021</div>
+                    <div class="text-2xl lg:text-4xl">10th Anniversary</div>
```
Comment (mattwang44) [inline pages/index.vue:24]: The title doesn't align with the current design.

```suggestion
                    <div class="text-3xl lg:text-5xl">PyCon Taiwan 10th Anniversary</div>
```

File: components/core/footer/Footer.vue (hunk 1 above)
Comment (flynn-hou) [reply components/core/footer/Footer.vue]: updated

File: components/core/footer/Footer.vue (hunk 2 above)
Comment (flynn-hou) [reply components/core/footer/Footer.vue]: updated

**Comment by mattwang44** [comment]
[RWD](https://trello.com/c/wSM2MMNe/37-rwd)

**Review by mattwang44 — APPROVED** [review]
LGTM

**Pull request merged**